### PR TITLE
[WIP] APIs empty_like and allocate_like for both cudf::column and cudf::table

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
APIs empty_like and allocate_like for both `cudf::column` and `cudf::table`.

### CODE REVIEW

File: cpp/src/copying/copy.cpp [hunk 1]
```diff
@@ -9,169 +9,186 @@
...
 #include <algorithm>
 
 namespace cudf
 {
+namespace experimental
+{
+namespace detail
+{
+
+inline mask_state should_allocate_mask(mask_allocation_policy mask_alloc, bool mask_exists) {
```
Comment (cwharris) [inline cpp/src/copying/copy.cpp]: just testing my understanding...

would it be appropriate to chance `bool mask_exists` to something like `mask_state existence` ?

Comment (rgsl888prabhu) [reply cpp/src/copying/copy.cpp]: I will try to take care of it

**Comment by rgsl888prabhu** [comment]
Closing this PR as we are planning to create two PRs, one with moving things to legacy and another to introduce new PRs.

File: cpp/src/copying/copy.cpp (hunk 1 above)
Comment (rgsl888prabhu) [reply cpp/src/copying/copy.cpp]: @cwharris `mask_state` is not associated with column and to get back the mask_state we have to again use `nullable()` and 'has_nulls()' to map it back to one of the mask_state. Which just increase steps and doesn't add much to code.

**Pull request closed without merging**